source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
 
         return view;
     }
+
+    public static boolean tryParseProtocolHandlerArguments(final String[] applicationArgs) {
+
+        if (applicationArgs == null) {
+            return false;
+        }
+
+        boolean found = false;
+
+        for (final String arg : applicationArgs) {
+            if (found) {
+                final URI uri = URIUtils.newURI(arg);
+
+                if (!uri.getScheme().equalsIgnoreCase(PROTOCOL_HANDLER_SCHEME)) {
+                    found = false;
+                    continue;
+                }
+
+                boolean repoUrlAvailable = false;
+                final String[] queryItems = uri.getQuery().split("&"); //$NON-NLS-1$
```

Comment:
I don't think uri.getQuery() is guaranteed not to return null. You may need a null check here unless you know your URL will always have parameters.